lib/engine.js

```diff
@@ -38,13 +38,20 @@ Engine = function(initializer) {
   engine = this;
   isStarted = false;
   engine.entityList = Lill.attach({});
-  engine.addEntity = function(components) {
-    var entity;
-    entity = Entity(components);
+  engine.addEntity = function(entity) {
+    if (!entity) {
+      entity = new Entity;
```

Comment:
Yeah, was thinking about this as well. Using `buildEntity` already creates a new Entity, so the `if not entity` check isn't necessary in `addEntity`.